### PR TITLE
make `rpcchainvm` die on SIGINT

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

Makes signal handling same as avalanchego

## How this works

See comments below

## How this was tested

n/a

### CODE REVIEW

File: vms/rpcchainvm/vm.go [hunk 1]
```diff
@@ -43,20 +43,11 @@ func Serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption
 			fmt.Println("vm server: graceful termination success")
 		}()
 
-		for {
```
Comment (danlaine) [inline vms/rpcchainvm/vm.go:46]: We return on receipt of any any of the handled signals or context cancellation so no need for loop

File: vms/rpcchainvm/vm.go [hunk 2]
```diff
@@ -43,20 +43,11 @@ func Serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption
...
-			select {
-			case s := <-signals:
-				switch s {
```
Comment (danlaine) [inline vms/rpcchainvm/vm.go:49]: We only handle `SIGINT` and `SIGTERM` and both are handled the same way so no need for switch

**Review by anusha-ctrl — APPROVED** [review]
Tested locally on ANR. SIGTERM sent to avalanchego processes ends all chains gracefully.

**Comment by StephenButtolph** [comment]
> Makes signal handling same as avalanchego

I don't think the goal was for this logic to be the same as avalanchego.

**Review by StephenButtolph — CHANGES_REQUESTED** [review]
I'm unable to replicate the bug that this is supposed to be fixing. Can we add some more detailed testing instructions?

**Comment by hexfusion** [comment]
I believe the problem with supporting `SIGINT` is ctrl-c sends `SIGINT` to both the main process and the subprocess. This results in the same condition as systemd. We ran into this bug on the first iteration of go-plugin removal. 

ref https://github.com/ava-labs/avalanchego-internal/pull/2682

**Comment by hexfusion** [comment]
cc @ceyonur

**Comment by StephenButtolph** [comment]
Closing this - as it seems like it would re-introduce a regression

**Pull request closed without merging**